pygmt/helpers/utils.py

```diff
@@ -387,10 +387,10 @@ def data_kind(
     match data:
         case None if required:  # No data provided and required=True.
             kind = "empty"
-        case str() | pathlib.PurePath():  # One file.
+        case str() | os.PathLike():  # One file.
```

Comment:
A little unsure about this, [`os.PathLike`](https://docs.python.org/3/library/os.html#os.PathLike) is an abstract base class that only needs `__fspath__` to be implemented as far as I can tell, and that can return either `str` or `bytes`. GMT can accept `str`, but unsure about raw `bytes`.

Reply:
The GMT API does not support raw `bytes` as file paths. However, according to the Python documentation for `__fspath__`:

> The method should only return a str or bytest object, with a preference for str.

To my knowledge, no built-in Python classes return `bytes` from `__fspath__()`, so this case should be rare in practice. In earlier versions, such `bytes`-returning path-like objects are recognized as vectors; now, they are treated as file inputs instead.